File: Src/java/model/src/main/java/org/hl7/cql/model/InstantiationContext.java
package org.hl7.cql.model;

public interface InstantiationContext extends ResolutionContext {
    boolean isInstantiable(TypeParameter parameter, DataType callType);

    DataType instantiate(TypeParameter parameter);

    Iterable<SimpleType> getSimpleConversionTargets(DataType callType);

    Iterable<IntervalType> getIntervalConversionTargets(DataType callType);

    Iterable<ListType> getListConversionTargets(DataType callType);
}

File: Src/java/cql-to-elm/src/commonMain/kotlin/org/hl7/cql/model/InstantiationContext.kt
package org.hl7.cql.model

interface InstantiationContext {
    fun isInstantiable(parameter: TypeParameter, callType: DataType): Boolean

    fun instantiate(parameter: TypeParameter): DataType

    fun getSimpleConversionTargets(callType: DataType): List<SimpleType>

    fun getIntervalConversionTargets(callType: DataType): List<IntervalType>

    fun getListConversionTargets(callType: DataType): List<ListType>
}
